cli/lsp/config.rs

```diff
@@ -1095,7 +1095,15 @@ impl Config {
         continue;
       };
       let settings = self.workspace_settings_for_specifier(workspace_uri);
-      if settings.enable.unwrap_or_else(|| self.has_config_file()) {
```

Comment:
@nayeemrmn this previous condition seemed slightly wrong and caused some trouble in the tests. Is the new condition ok?

Reply:
Ah good find, but if `enable_paths` is specified then it should completely override `enable`.

```rust
if settings.enable_paths.as_ref().map(|p| !p.is_empty()).unwrap_or_else(|| previous_condition) { ... }
```